### DISCUSSION TITLE
Highlight - support for additional languages

### DISCUSSION THREAD

**Discussion by thedaxshepherd** [chipzoller/hugo-clarity#275] [Q&A]
First, I would like to say I love the look of this theme and how it handles code blocks. 

But I am trying to add support for additional languages such as DAX, Power Query and Powershell. My first attempt was to update the Highlight.js file with one downloaded from their site and replace the one in assets/js/highlight.js. This caused it to break. I take it because the original file is a couple of major versions behind and it looks like the format is different. 

DAX is not available as part of the download so I even downloaded the highlight source and built the file with a DAX.js source. But this also generates a file that is not interchangeable with the highlight.js file in the theme. 

What I did get to work is using Prism.js for the couple extra languages I need and adding it to the custom.css and custom.js files. Since it uses code and line-number classes I renamed them so they do not clobber each other. (I know at this point people are shaking their heads. This is not optimal but I try to get something working and then research how to properly handle this case). This solution work but I know it is first inefficient since I am adding bloat to the js and css that has to be downloaded. Also from a user's perspective, they have two different code blocks and how they work. Not a big deal since I probably wouldn't mix languages in the same post. 

Has anyone been able to update the highlight.js file with a newly downloaded one? Also if I do go the route of using Prism.js - any suggestions on ways I should look at to improve how I am going about it. I am open to all suggestions and this is outside my skillset but trying to get up to speed as quickly as possible. If anything this shows how much I like the theme because I want to use it even if it takes more work.

**Comment by chipzoller** [comment]
@onweru 's work here

**Reply by thedaxshepherd** [reply]
Just wondering if there was any update to this.

**Reply by chipzoller** [reply]
Will have to wait for an update from @onweru on this one.

**Comment by onweru** [comment]
@thedaxshepherd, I will review this over the weekend and communicate.

**Reply by thedaxshepherd** [reply]
Any update on this?

**Comment by onweru** [comment]
@thedaxshepherd, I confirmed that highlight js [doesn't have support](https://highlightjs.org/download/) for DAX. While I was at it, I remembered that when I added highlight js, I selected only a few languages e.g `yaml` ... basically only the ones 1) we needed at the time 2) were not already supported by Hugo's syntax highlighter. The intention was to keep the build light.

I also observed that Prixm js indeed supported all the languages you listed. I agree it is better for compatibility and would very much like to switch to it. However, even if we switch to it, we will only include a minimal build of the prixm script (the full build is 556kb). And provide a way for users to override it with a build containing the languages they miss.

**Reply by onweru** [reply]
In retrospect, I think prism is a pain to implement https://discourse.gohugo.io/t/problems-with-hugo-and-prism-js/17357, dropping the idea :)

**Comment by thedaxshepherd** [comment]
Thank you for looking at this. I agree that implementing Prism would not be a good use of resources.
I also understand that the audience for DAX is a small percentage but should grow
as more poeple use Power BI.

I like Highlight.js because as you mentioned you can select only the languages you want and download the
file already built. Optionally you can also download the source and build the file using languages using source
files like DAX, etc that are not included in the original list of languages.

The issue I am having is when I download the newest file (VER 11+) and replace the highlight.js file I get a 
EXECUTE-AS-TEMPLATE failure to parse message:
![HighlightJS11_Error](https://user-images.githubusercontent.com/60486315/170319730-715aeaf3-ee21-40c4-80a8-5e68f7032905.png)

I know that the currently used file is EOL and needs to be upgraded at some point:
https://github.com/highlightjs/highlight.js/blob/main/SECURITY.md

Can upgrading highlight.js to the newest version be something on the roadmap. I was trying to see if 
I could fix the issue myself but so far I have not found a solution while researching it. Again I appreciate all the help.

**Reply by thedaxshepherd** [reply]
I appreciate the time in looking into this. For now, I have decided to go a different route so I can get up and running and focus on content. I tried various methods of using both syntax highlighters or converting the syntax for languages that aren't supported to HTML manually and nothing really felt right. I still love this theme.
